corehq/apps/reports_core/filters.py

```diff
@@ -261,13 +264,33 @@ def __init__(self, name, field, datatype, label, show_all, url_generator, choice
         self.url_generator = url_generator
         self.choice_provider = choice_provider
 
+    def context(self, values, lang=None):
+        context = super(DynamicChoiceListFilter, self).context(values, lang)
+        context['value'] = self._values_to_dict(values)
+        return context
+
+    def _values_to_dict(self, values):
+        """Some values are returned as Choice objects which need to be converted to
+        dicts to be displayed properly
+        """
+        if values:
+            return [
+                dict(value._asdict()) if isinstance(value, Choice) else value
+                for value in values
+            ]
```

Comment:
It would be nice if this function always returned a list (empty if there are no values) rather than list or None. It often makes it easier to reason about a function when it only returns a single type.

Also, the name `_values_to_dict` implies that it returns a dict. Would be nice if it were clear that it returns a list.